### PR TITLE
Add test for nested RecordEvent.

### LINKED ISSUE #8785: Need to add unittest for nested paddle::platform::RecordEvent
The `paddle::platform::RecordEvent` is originally designed to support nested profiling. But there is no unittest for it. As a result, some modification may accidentally disable, causing errors like #8753 . Thus, we need to add unittest for this usage.

### PR DESCRIPTION
**Pull request by Xreki**
Fix #8785

### CODE REVIEW

File: paddle/fluid/platform/profiler_test.cc [hunk 1]
```diff
@@ -100,6 +100,36 @@ TEST(RecordEvent, RecordEvent) {
...
+  LOG(INFO) << "Usage 3: nested RecordEvent:";
+  for (int i = 1; i < 5; ++i) {
+    std::string name = "ano_evs_op_" + std::to_string(i);
+    LOG(INFO) << name;
+    RecordEvent record_event(name, dev_ctx);
+    int counter = 1;
+    while (counter != i * 1000) counter++;
+    {
+      std::string nested_name = "nested_ano_evs_op_" + std::to_string(i);
+      LOG(INFO) << nested_name;
```
Comment (kuke) [inline paddle/fluid/platform/profiler_test.cc]: Can remove this line and line 120 for clean output.

**Review by kuke — COMMENTED** [review]
Almost LGTM. And can you add log info for usage 1 and usage 2?

File: paddle/fluid/platform/profiler_test.cc (hunk 1 above)
Comment (Xreki) [reply paddle/fluid/platform/profiler_test.cc]: Oh, sorry, I forgot that ...

File: paddle/fluid/platform/profiler_test.cc [hunk 2]
```diff
@@ -100,6 +100,33 @@ TEST(RecordEvent, RecordEvent) {
...
+   *     code to be analyzed
+   *     ...
+   *   }
+   * }
+   */
+  for (int i = 1; i < 5; ++i) {
+    std::string name = "ano_evs_op_" + std::to_string(i);
+    RecordEvent record_event(name, dev_ctx);
+    int counter = 1;
+    while (counter != i * 1000) counter++;
```
Comment (qingqing01) [inline paddle/fluid/platform/profiler_test.cc]: reduce the count in while?

Comment (Xreki) [reply paddle/fluid/platform/profiler_test.cc]: Done.

**Review by kuke — APPROVED** [review]
LGTM

**Pull request merged**